src/interaction/keybindingmanager.cpp

```diff
@@ -0,0 +1,175 @@
+#include <ghoul/logging/logmanager.h>
+
+#include <glm/gtx/string_cast.hpp>
+#include <ghoul/glm.h>
+
+#include <fstream>
+
+namespace {
+    const char* _loggerCat = "KeyBindingManager";
+
+    const char* MainTemplateFilename = "${OPENSPACE_DATA}/web/keybindings/main.hbs";
+    const char* KeybindingTemplateFilename = "${OPENSPACE_DATA}/web/keybindings/keybinding.hbs";
+    const char* JsFilename = "${OPENSPACE_DATA}/web/keybindings/script.js";
+} // namespace
+
+#include "keybindingmanager_lua.inl"
+
+namespace openspace {
+namespace interaction {
+
+KeyBindingManager::KeyBindingManager()
+    : DocumentationGenerator(
+        "Documentation",
+        "keybindings",
+        {
+            { "keybindingTemplate",  KeybindingTemplateFilename },
+            { "mainTemplate", MainTemplateFilename }
+        },
+        JsFilename
+    )
+{ }
+
+KeyBindingManager::~KeyBindingManager()
+{ }
+
+void KeyBindingManager::initialize()
+{ }
+
+void KeyBindingManager::deinitialize()
+{ }
+
+void KeyBindingManager::keyboardCallback(Key key, KeyModifier modifier, KeyAction action) {
+    if (action == KeyAction::Press || action == KeyAction::Repeat) {
+        // iterate over key bindings
+        auto ret = _keyLua.equal_range({ key, modifier });
+        for (auto it = ret.first; it != ret.second; ++it) {
+            auto remote = it->second.synchronization ?
+                scripting::ScriptEngine::RemoteScripting::Yes :
+                scripting::ScriptEngine::RemoteScripting::No;
+
+            OsEng.scriptEngine().queueScript(it->second.command, remote);
+        }
+    }
+}
+
+void KeyBindingManager::resetKeyBindings() {
+    _keyLua.clear();
+}
+
+void KeyBindingManager::bindKeyLocal(Key key, KeyModifier modifier,
+                                      std::string luaCommand, std::string documentation)
+{
+    _keyLua.insert({
+        { key, modifier },
+        { std::move(luaCommand), Synchronized::No, std::move(documentation) }
+    });
+}
+
+void KeyBindingManager::bindKey(Key key, KeyModifier modifier,
+                                 std::string luaCommand, std::string documentation)
+{
+    _keyLua.insert({
+        { key, modifier },
+        { std::move(luaCommand), Synchronized::Yes, std::move(documentation) }
+    });
+}
+    
+std::string KeyBindingManager::generateJson() const {
+    std::stringstream json;
+    json << "[";
+    bool first = true;
+    for (const auto& p : _keyLua) {
+        if (!first) {
+            json << ",";
+        }
+        first = false;
+        json << "{";
+        json << "\"key\": \"" << std::to_string(p.first) << "\",";
+        json << "\"script\": \"" << p.second.command << "\",";
+        json << "\"remoteScripting\": " << (p.second.synchronization ? "true," : "false,");
+        json << "\"documentation\": \"" << p.second.documentation << "\"";
+        json << "}";
+    }
+    json << "]";
+    
+    std::string jsonString = "";
+    for (const char& c : json.str()) {
```

Comment:
I think this should be factored out since it is going to be used in different places...


wait.. i think it already is somewhere